### PR TITLE
Adding get cause and get permission logic

### PR DESCRIPTION
**Pull request by FlappiTomic**
Signed-off-by: Tom Zhang <tom.yczhang@gmail.com>

### CODE REVIEW

File: src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy [hunk 1]
```diff
@@ -1431,6 +1433,19 @@ class GenericPipeline extends Pipeline {
...
+        this.isAuthorizedUser = false
+        if (causeID == PipelineConstants.USERID_CAUSE_ID) {
+            this.isAuthorizedUser = true
+        } else if (causeID == PipelineConstants.BRANCHEVENT_CAUSE_ID || causeID == PipelineConstants.BRANCHINDEXING_CAUSE_ID) {
+            if (this.changeInfo.isPullRequest) {
+                this.isAuthorizedUser = isPRAuthorizedUser()
+            }
+        }
+        if (!this.isAuthorizedUser) {
+            preSetupException = new PackagingStageException("this.isAuthorizedUser is not true, user not authorized to run build", this.isAuthorizedUser)
```
Comment (jackjia-ibm) [inline src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy:1446]: We can say automatic packaging step for non-committers on z/OS is disabled.

File: src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy [hunk 2]
```diff
@@ -1431,6 +1433,19 @@ class GenericPipeline extends Pipeline {
         //     preSetupException = new PackagingStageException("arguments.localWorkspace is not defined for packagingGeneric", arguments.name)
         // }
 
+        Integer causeID = Pipeline.build.getCause()
```
Comment (jackjia-ibm) [inline src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy:1436]: should be `this.build.getCause()`?

File: src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy [hunk 3]
```diff
@@ -1519,6 +1534,20 @@ class GenericPipeline extends Pipeline {
...
+        
+        String prNumberString = "${this.changeInfo.pullRequestId}"   // this will be PR number
+        int prNumber = prNumberString as Integer   // convert to int
+        def user = this.github.getPullRequestUser(prNumber)
+        def allowed = this.github.isUserWriteCollaborator(user)
+        if (allowed) {
+            isAuthorized = true
+        }
+        
+        return isAuthorized
```
Comment (jackjia-ibm) [inline src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy:1548]: we can just do `return this.github.isUserWriteCollaborator(user)`

File: src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy (hunk 1 above)
Comment (FlappiTomic) [reply src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy:1446]: ok

**Review by jackjia-ibm — APPROVED** [review]
Looks great. Thanks Tom!

**Pull request merged**